opal/mca/common/ofi/common_ofi.c

```diff
@@ -627,10 +627,10 @@ static bool is_near(pmix_device_distance_t *distances,
                 continue;
 
             for (i = 0; i < num_distances; i++) {
-                char lsguid[256], lnguid[256];
+                char lsguid[20], lnguid[20];
                 int ret;
 
-                ret = sscanf(distances[i].uuid, "fab://%256s::%256s", lnguid, lsguid);
+                ret = sscanf(distances[i].uuid, "fab://%19::%19", lnguid, lsguid);
```

Comment:
not sure I follow how that helps. Can you explain?

Reply:
Currently the execution flow always exits at L635 because of the string format, causing `is_near` to return on L658, even though the nic is "near".